tests/unit/torch/blocks/test_experts.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+import torch
+
+import merlin.models.torch as mm
+from merlin.models.torch.blocks.experts import (
+    CGCBlock,
+    ExpertGateBlock,
+    MMOEBlock,
+    PLEBlock,
+    PLEExpertGateBlock,
+)
+from merlin.models.torch.utils import module_utils
+
+dict_inputs = {"experts": torch.rand((10, 4, 5)), "shortcut": torch.rand((10, 8))}
+
+
+class TestExpertGateBlock:
+    @pytest.fixture
+    def expert_gate(self):
+        return ExpertGateBlock(num_experts=4)
+
+    def test_requires_dict_input(self, expert_gate):
+        with pytest.raises(RuntimeError, match="ExpertGateBlock requires a dictionary input"):
+            expert_gate(torch.rand((10, 5)))
+
+    def test_forward_pass(self, expert_gate):
+        result = module_utils.module_test(expert_gate, dict_inputs)
+        assert result.shape == (10, 5)
+
+
+class TestMMOEBlock:
+    def test_init(self):
+        mmoe = MMOEBlock(mm.MLPBlock([2, 2]), 2)
+
+        assert isinstance(mmoe, MMOEBlock)
+        assert isinstance(mmoe[0], mm.ShortcutBlock)
+        assert len(mmoe[0][0].branches) == 2
+        for i in range(2):
+            assert mmoe[0][0][str(i)][1].out_features == 2
+            assert mmoe[0][0][str(i)][3].out_features == 2
+        assert isinstance(mmoe[0][0].post[0], mm.Stack)
+        assert isinstance(mmoe[1], ExpertGateBlock)
+        assert mmoe[1][0][0].out_features == 2
+
+    def test_init_with_outputs(self):
+        outputs = mm.ParallelBlock({"a": mm.BinaryOutput(), "b": mm.BinaryOutput()})
+        outputs.prepend_for_each(mm.MLPBlock([2]))
+        outputs.prepend(MMOEBlock(mm.MLPBlock([2, 2]), 2, outputs))
```

Comment:
The API is very clean!. But building MTL models from the output to the input kind of works, but might be counter-intuitive. Will that be the pattern for MTL models?
But I think I understand the challenges of building such MMOE models from the inputs, as gates depend on # experts and number of gates and towers depend on numbers of outputs.

Reply:
Yeah, it might make sense to add a `MMOEOutputs` or something that does this under-the-hood.